### PR TITLE
Added domains api

### PR DESCRIPTION
**Pull request by avbel**

### CODE REVIEW

File: lib/domain.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			path   : "domains",
+			method : "GET",
+			qs     : { size : 100 }
+		})
+		.then(function (response) {
+			return response.body;
+		})
+		.asCallback(callback);
+	};
+
```
Comment (dtolb) [inline lib/domain.js:66]: For `list` we're building out a `next` link here. Check out how [applications](https://github.com/bandwidthcom/node-bandwidth/blob/2.0-preview/lib/application.js#L27) paginate.

From @fuchsnj 

> We do not want to allow people to use the page param. If someone needs the 2nd page, they should follow the 'next' link.
> Using the page parameters has a significant performance impact on Catapult and we are starting to migrate to something else that is faster. If you use the 'next' link we can switch to a different paging method without you knowing.

Comment (avbel) [inline lib/domain.js:66]: Domains api doesn't support pagination (at least this info is missing in docs). By default GET /domains returns only 25 domains. With param `size` = 100 it returns all domains (max. allowed domains count is 100). Param `page` in query does nothing. So I don't see any sense to support `next` here.

**Comment by dtolb** [comment]
@nguyer @fuchsnj @nbragdon @a-type Looks good to me... +1?

File: docs/api.md [hunk 2]
```diff
@@ -263,6 +266,62 @@ Gets a list of active and historic calls you made or received.
...
+
+<a name="new_Domain_new"></a>
+
+### new Domain()
+Domain
+
+<a name="Domain+create"></a>
+
+### domain.create(params, [callback]) ⇒ <code>[CallResponse](#CallResponse)</code>
+Create a new voice call
```
Comment (a-type) [inline docs/api.md]: This seems wrong. Also, almost all the links here are to `CallResponse`. Is that intentional? Maybe I don't understand what a domain does in the SDK.

**Comment by dtolb** [comment]
Ah, I see what happened with the jsdoc. 

The domains.js needs a domain response. See how [applications](https://github.com/bandwidthcom/node-bandwidth/blob/2.0-preview/lib/application.js#L163) returns on `list`

**Comment by a-type** [comment]
K, seems good

**Pull request merged**